app/controllers/PromoLandingPage.scala

```diff
@@ -1,61 +1,108 @@
 package controllers
 import actions.CommonActions._
+import com.gu.i18n.Country
+import com.gu.memsub.Digipack
 import com.gu.memsub.promo.Formatters.PromotionFormatters._
 import com.gu.memsub.promo.Promotion._
 import com.gu.memsub.promo._
 import com.netaporter.uri.dsl._
 import configuration.Config
+import controllers.SessionKeys.PromotionTrackingCode
 import filters.HandleXFrameOptionsOverrideHeader
-import model.DigitalEdition
+import model.DigitalEdition.UK
 import play.api.data.{Form, Forms}
 import play.api.libs.json.Json
 import play.api.mvc._
 import play.twirl.api.Html
 import services.TouchpointBackend
-import utils.Tracking.internalCampaignCode
+import utils.Tracking._
+import views.html.promotion._
 import views.support.PegdownMarkdownRenderer
 
+import scala.concurrent.ExecutionContext.Implicits.global
+
 object PromoLandingPage extends Controller {
 
-  val tpBackend = TouchpointBackend.Normal
-  lazy val catalog = tpBackend.catalogService.unsafeCatalog
-  val edition = DigitalEdition.UK
+  private val tpBackend = TouchpointBackend.Normal
+  private val catalog = tpBackend.catalogService.unsafeCatalog
+
+  private val digitalPackRatePlanIds = catalog.digipack.toList.map(_.id).toSet
+  private val allPaperPackages = catalog.delivery.list ++ catalog.voucher.list
+  private val paperOnlyPackageRatePlanIds = allPaperPackages.filter(_.charges.benefits.list.contains(Digipack)).map(_.id).toSet
+  private val paperPlusPackageRatePlanIds = allPaperPackages.filterNot(_.charges.benefits.list.contains(Digipack)).map(_.id).toSet
+  private val guardianWeeklyRatePlanIds = catalog.weeklyZoneA.toList.map(_.id).toSet ++ catalog.weeklyZoneC.toList.map(_.id).toSet
 
   private def isActive(promotion: AnyPromotion): Boolean = {
     val isTest = tpBackend.environmentName != "PROD"
-    isTest || (promotion.starts.isBeforeNow && !promotion.expires.exists(_.isBeforeNow))
+    isTest || (
```

Comment:
kind of feel like this should be in the promotion itself, as other parts of the code would need to do this too